### PR TITLE
download panel normally show "downloading" and download completely even download panel is opened before the "downloading" toast show.

### PR DESCRIPTION
**Pull request by WeiWW**
as title

### CODE REVIEW

**Comment by Conjuror** [comment]
By spec, it should be snackbars with action.

File: app/src/main/java/org/mozilla/focus/components/DownloadCompleteReceiver.java [hunk 1]
```diff
@@ -57,15 +59,21 @@ public void onReceive(Context context, Intent intent) {
...
                                 downloadId,
                                 downloadedFile,
                                 mediaType);
                     }
+
+                    String completedStr = context.getString(R.string.download_completed);
+                    Toast.makeText(context,fileName+" "+completedStr,Toast.LENGTH_SHORT).show();
                 }
             }
+        }else {
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/focus/components/DownloadCompleteReceiver.java]: Last mile: Let's also notify Adapter with a new delete item method to deal with the case when the user cancels with Download Manager Toast...well this is an edge case so I guess we can skip this for now. Once we merge this I will file another bug for this and I might triage it as P2.

**Comment by mTwTm** [comment]
As always, it is generally a good idea to split changes you have made into different commits and different PRs, this would not only make reviews easier, but also give me chances to merge part of the otherwise-too-large pr and speed up our test cycle.

File: app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java [hunk 2]
```diff
@@ -160,21 +156,31 @@ public void delete(Long downloadId, AsyncDeleteListener listener) {
...
         mQueryHandler.startQuery(TOKEN, listener, Uri.parse(uri), null, null, null, null);
     }
 
-    public void queryCertainId(Long downloadId,AsyncQueryListener listener){
+    public void queryByDownloadId(Long downloadId, AsyncQueryListener listener){
         String uri = Download.CONTENT_URI.toString();
         mQueryHandler.startQuery(TOKEN,listener,Uri.parse(uri),null,Download.DOWNLOAD_ID+"==?",new String[] {Long.toString(downloadId)},null);
     }
 
+    public void queryByRowId(Long rowId,AsyncQueryListener listener){
```
[the remark this answers is not in the document]
Comment (WeiWW) [reply app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java]: Because the downloadID maybe change during moving file path, the row id is permanent.

File: app/src/main/java/org/mozilla/focus/provider/DownloadInfoProvider.java [hunk 3]
```diff
@@ -45,7 +45,7 @@ public Cursor query(@NonNull Uri uri, @Nullable String[] projection, @Nullable S
         }
 
         SQLiteDatabase sqLiteDatabase = mDbHelper.getReadableDB();
-        Cursor cursor = queryBuilder.query(sqLiteDatabase,projection,selection,selectionArgs,null,null,sortOrder,getLimitParam(uri.getQueryParameter("offset"), uri.getQueryParameter("limit")));
+        Cursor cursor = queryBuilder.query(sqLiteDatabase, projection, selection, selectionArgs, null, null, sortOrder, getLimitParam(uri.getQueryParameter("offset"), uri.getQueryParameter("limit")));
```
[the remark this answers is not in the document]
Comment (WeiWW) [reply app/src/main/java/org/mozilla/focus/provider/DownloadInfoProvider.java]: Sorry, I'll notice next time.

**Comment by mTwTm** [comment]
Please rebase to current implementation before moving on.

File: app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java [hunk 4]
```diff
@@ -30,13 +30,14 @@
...
     public static final String ROW_ID = "row id";
+    public static final String INSERT_SUCCESS = "insert successful";
     private static final int TOKEN = 2;
     private static DownloadInfoManager sInstance;
     private static Context mContext;
     private static DownloadInfoQueryHandler mQueryHandler;
 
     public static DownloadInfoManager getInstance() {
-        if (sInstance == null) {
+        if (sInstance == null){
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java]: Please review yourself for unnecessary changes before sending pull requests.

File: app/src/main/java/org/mozilla/focus/fragment/DownloadsFragment.java [hunk 5]
```diff
@@ -47,20 +49,22 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
-                    DownloadInfoManager.getInstance().queryByDownloadId(id, new DownloadInfoManager.AsyncQueryListener() {
-                        @Override
-                        public void onQueryComplete(List downloadInfoList) {
-                            for (int i=0;i<downloadInfoList.size();i++){
-                                DownloadInfo downloadInfo = (DownloadInfo) downloadInfoList.get(i);
-                                mDownloadListAdapter.updateItem(downloadInfo);
-                            }
-                        }
-                    });
+                if (intent.getAction().equals(DownloadManager.ACTION_DOWNLOAD_COMPLETE)){
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/focus/fragment/DownloadsFragment.java]: Please select this block and get it properly indented with Android Studio.

File: app/src/main/java/org/mozilla/focus/fragment/DownloadsFragment.java [hunk 6]
```diff
@@ -65,13 +69,14 @@ public void onQueryComplete(List downloadInfoList) {
     public void onResume() {
         super.onResume();
         IntentFilter intentFilter = new IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE);
-        getActivity().registerReceiver(mDownloadReceiver, intentFilter);
+        intentFilter.addAction(DownloadInfoManager.INSERT_SUCCESS);
```
[the remark this answers is not in the document]
Comment (mTwTm) [reply app/src/main/java/org/mozilla/focus/fragment/DownloadsFragment.java]: This is not fixed yet. Don't share the IntentFilter

File: app/src/main/java/org/mozilla/focus/provider/DownloadInfoProvider.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mTwTm) [reply app/src/main/java/org/mozilla/focus/provider/DownloadInfoProvider.java]: Please move these into a separate commit if you have time :)
